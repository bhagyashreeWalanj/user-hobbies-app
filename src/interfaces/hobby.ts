export enum PassionLevel {
    Low = 'Low',
    Medium = 'Medium',
    High = 'High',
    VeryHigh = 'Very-High',
  }
  
  export interface IHobby {
    id: string
    hobby: string
    passion: PassionLevel | string
    year: string
  }