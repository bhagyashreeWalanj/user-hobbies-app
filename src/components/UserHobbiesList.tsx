import styles from './../styles/UserHobbiesList.module.scss'
import { IHobby } from './../interfaces/hobby'
import { MdDelete } from 'react-icons/md'

// An Interface for User Hobbies List
export interface UserHobbiesListProps {
  hobbies: IHobby[]
  removeHobby(hobbyId: string): void
}

const UserHobbiesList = ({ hobbies, removeHobby }: UserHobbiesListProps) => {
  return (
    <div className={styles.container}>
      {hobbies.length === 0 ? (
        <div className={styles.noDataDiv}>No data Found!</div>
      ) : (
        hobbies.map((item) => {
          return (
            <div key={item.id} className={styles.item}>
              <span>{item.passion}</span>
              <span>{item.hobby}</span>
              <span>Since {item.year}</span>
              <MdDelete
                className={styles.deleteBtn}
                onClick={() => removeHobby(item.id)}
                name="delete"
                id={`deleteBtn_${item.id}`}
              />
            </div>
          )
        })
      )}
    </div>
  )
}

export default UserHobbiesList
