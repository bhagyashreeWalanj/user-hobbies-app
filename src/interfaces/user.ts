import { IHobby } from './hobby'

// Define the User type
export interface IUser {
    id: string,
      name: string,
      hobbies: IHobby[]
  }