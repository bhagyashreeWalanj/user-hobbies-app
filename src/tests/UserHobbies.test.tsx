import React from 'react';
import { mount, shallow, render } from 'enzyme';
import { IUser } from '../interfaces/user';
import UserHobbiesList from '../components/UserHobbiesList';
import { IHobby } from '../interfaces/hobby';


describe('Testing User Hobbies Component', () => {
  const addHobby = () => {}
  const removeHobby = jest.fn();
  const user : IUser ={
    id: '1',
    name: '22',
    hobbies: [{
      id: '2',
      hobby: 'Playing Cricket',
      passion: 'Low',
      year: '1990'
    }]
  }
  const hobbies:IHobby[]=[
    {
      id: '1',
      hobby: 'Playing Cricket',
      passion: 'Low',
      year: '1990'
    },
    {
      id: '2',
      hobby: 'Playing Cricket',
      passion: 'Low',
      year: '1990'
    }
  ]
  const wrapper = shallow(
    <UserHobbiesList hobbies={hobbies} removeHobby={removeHobby} />
  );
  
  it('should Renders user Hobbies Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should remove hobby by Id', () => {
    wrapper.find('#deleteBtn_1').simulate('click');
   expect(removeHobby).toBeCalledTimes(1);

  });

  
});




