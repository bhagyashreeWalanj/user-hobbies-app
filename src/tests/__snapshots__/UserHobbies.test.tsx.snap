// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User Hobbies Component should Renders user Hobbies Component 1`] = `
<div
  className="container"
>
  <div
    className="item"
    key="1"
  >
    <span>
      Low
    </span>
    <span>
      Playing Cricket
    </span>
    <span>
      Since 
      1990
    </span>
    <MdDelete
      className="deleteBtn"
      id="deleteBtn_1"
      name="delete"
      onClick={[Function]}
    />
  </div>
  <div
    className="item"
    key="2"
  >
    <span>
      Low
    </span>
    <span>
      Playing Cricket
    </span>
    <span>
      Since 
      1990
    </span>
    <MdDelete
      className="deleteBtn"
      id="deleteBtn_2"
      name="delete"
      onClick={[Function]}
    />
  </div>
</div>
`;
